### PR TITLE
feat: attribute filter with secondary label

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
- secondary label does not affect execution anymore
- primary label is always used for execution
- secondary label is stored outside of filter for UI only
- each AF in Insight has now localIdentifier
- secondary label references Af via localId
- displayForms now contain isPrimary, isDefault flags
- AF keeps its selection when secondary label switched
- change of secondary label does not recreate AF handler
- collectLabelElements can now filter by primary label

JIRA: LX-183
risk: high

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

File: libs/sdk-backend-tiger/src/convertors/fromBackend/CatalogConverter.ts [hunk 1]
```diff
@@ -46,13 +46,16 @@ const commonGroupableCatalogItemModifications =
 const tigerLabelToDisplayFormMd = (
     label: JsonApiLabelOutWithLinks,
     attributeRef: ObjRef,
+    defaultLabelId?: string,
```
[the remark this answers is not in the document]
Comment (ivan-nejezchleb) [reply libs/sdk-backend-tiger/src/convertors/fromBackend/CatalogConverter.ts:49]: the same convertor is used also for geoLabels manipulation and there the default is not provided as it is not relevant

File: libs/sdk-model/src/execution/filter/factory.ts [hunk 2]
```diff
@@ -14,10 +14,27 @@ import {
...
+} from "../../objRef/index.js";
 import { IMeasure, isMeasure, measureLocalId } from "../measure/index.js";
 import { idRef, localIdRef } from "../../objRef/factory.js";
 import { DateAttributeGranularity } from "../../base/dateGranularities.js";
+import SparkMD5 from "spark-md5";
+import { sanitizeLocalId } from "../../sanitizeLocalId.js";
+
+function generateLocalId(prefix: string, objRef: ObjRef, inObject: IAttributeElements): string {
+    const hasher = new SparkMD5();
+    hasher.append(JSON.stringify(inObject));
```
[the remark this answers is not in the document]
Comment (ivan-nejezchleb) [reply libs/sdk-model/src/execution/filter/factory.ts:33]: I used same code as for generating other missing localIdentifiers, eg. libs/sdk-model/src/execution/measure/factory.ts

**Comment by ivan-nejezchleb** [comment]
ok to test

**Comment by ivan-nejezchleb** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2667/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by ivan-nejezchleb** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2668/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by ivan-nejezchleb** [comment]
extended test - tiger-cypress - isolated

**Comment by ivan-nejezchleb** [comment]
extended test - tiger-cypress - isolated

**Comment by ivan-nejezchleb** [comment]
extended test - tiger-cypress - isolated

**Comment by ivan-nejezchleb** [comment]
extended test - tiger-cypress - integrated

**Pull request merged**